serverless.yml

```
@@ -56,7 +56,29 @@ resources:
         Action: lambda:InvokeFunction
         Principal: apigateway.amazonaws.com
 
-  Outputs:
-    ApiUrl:
-      Description: "API URL"
-      Value: !Join ['', [ "https://", Ref: ApiGatewayRestApi, ".execute-api.", Ref: AWS::Region, ".amazonaws.com/${self:provider.stage}/" ]]
+    # Setup the application's domain name as a custom domain name for API Gateway, then map it to the application's API
+    ApiGatewayCustomDomainName:
+      Type: AWS::ApiGateway::DomainName
+      Properties:
+        RegionalCertificateArn: ${env:SSL_CERTIFICATE_ARN}
+        DomainName: ${env:DOMAIN_NAME}
+        EndpointConfiguration:
+          Types:
+            - REGIONAL
+
+    BasePathMapping:
+      Type: AWS::ApiGateway::BasePathMapping
+      DependsOn: [ApiGatewayDeployment]
+      Properties:
+        BasePath: ''
+        DomainName: !Ref ApiGatewayCustomDomainName
+        RestApiId: !Ref ApiGatewayRestApi
+        Stage: ${self:provider.stage}
+
+    # DNS CNAME mapping the application's domain name to the behind-the-scenes domain name for the API
+    AppCNAME:
```

It might be worth adding a comment that references the [AWS DNS record wiki page](https://wiki.york.ac.uk/display/AWS/AWS%3A+Application+DNS+Records) - I think an incoming developer might struggle to understand what's happening here without it.
```suggestion
    # DNS CNAME mapping the application's domain name to the behind-the-scenes domain name for the API
    # See https://wiki.york.ac.uk/display/AWS/AWS%3A+Application+DNS+Records
    AppCNAME:
```